crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_dict_comprehension_for_iterable.rs

```diff
@@ -99,6 +99,10 @@ pub(crate) fn unnecessary_dict_comprehension_for_iterable(
 
         let binding = checker.semantic().binding(id);
 
+        if binding.kind.is_builtin() {
```

Comment:
Oh wow, great investigation! Could you more-or-less copy and paste your PR summary as a comment here? It's otherwise unclear why we have to special-case builtin bindings 😄